tower-http/src/compression/future.rs

```diff
@@ -21,14 +25,40 @@ pub struct ResponseFuture<F> {
 
 impl<F, B, E> Future for ResponseFuture<F>
 where
-    F: Future<Output = Result<http::Response<B>, E>>,
+    F: Future<Output = Result<Response<B>, E>>,
     B: Body,
 {
-    type Output = Result<http::Response<CompressionBody<B>>, E>;
+    type Output = Result<Response<CompressionBody<B>>, E>;
+
+    #[allow(unreachable_code, unused_mut, unused_variables)]
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let res = ready!(self.as_mut().project().inner.poll(cx)?);
+
+        let (mut parts, body) = res.into_parts();
+
+        let body = match self.encoding {
+            #[cfg(feature = "compression-gzip")]
+            Encoding::Gzip => CompressionBody(BodyInner::Gzip(WrapBody::new(body))),
+            #[cfg(feature = "compression-deflate")]
+            Encoding::Deflate => CompressionBody(BodyInner::Deflate(WrapBody::new(body))),
+            #[cfg(feature = "compression-br")]
+            Encoding::Brotli => CompressionBody(BodyInner::Brotli(WrapBody::new(body))),
+            Encoding::Identity => {
+                return Poll::Ready(Ok(Response::from_parts(
+                    parts,
+                    CompressionBody(BodyInner::Identity(body)),
+                )))
+            }
+        };
+
+        parts.headers.remove(header::CONTENT_LENGTH);
+
+        parts.headers.insert(
+            header::CONTENT_ENCODING,
+            HeaderValue::from_str(self.encoding.to_str()).unwrap(),
```

Comment:
iirc `HeaderValue` is backed by a `Bytes` so maybe it makes sense to store this header value in the struct and make it a ref count bump?

Reply:
Good point! The header value is created from the request but we can use `HeaderValue::from_static` as the encodings are an enum.